code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/pushtemplate/BasicTemplateNotificationBuilder.kt

```diff
@@ -0,0 +1,500 @@
+            val expandedLayout = RemoteViews(packageName, R.layout.push_template_expanded)
+            if (pushImage != null) {
+                expandedLayout.setImageViewBitmap(R.id.expanded_template_image, pushImage)
+            }
+            smallLayout.setTextViewText(R.id.notification_title, titleText)
+            smallLayout.setTextViewText(R.id.notification_body, bodyText)
+            expandedLayout.setTextViewText(R.id.notification_title, titleText)
+            expandedLayout.setTextViewText(R.id.notification_body_expanded, expandedBodyText)
+            val actionUri = intentExtras.getString(PushTemplateConstants.IntentKeys.ACTION_URI)
+            val messageId =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.MESSAGE_ID) as String
+            val deliveryId =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.DELIVERY_ID) as String
+            val remindLaterTimestamp =
+                intentExtras.getLong(PushTemplateConstants.IntentKeys.REMIND_TS)
+            val remindLaterText =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.REMIND_LABEL)
+            val badgeCount = intentExtras.getInt(PushTemplateConstants.IntentKeys.BADGE_COUNT)
+            val visibility = intentExtras.getInt(PushTemplateConstants.IntentKeys.VISIBILITY)
+            val importance = intentExtras.getInt(PushTemplateConstants.IntentKeys.IMPORTANCE)
+            val channelId = intentExtras.getString(PushTemplateConstants.IntentKeys.CHANNEL_ID)
+            val notificationBackgroundColor = intentExtras.getString(
+                PushTemplateConstants.IntentKeys.NOTIFICATION_BACKGROUND_COLOR
+            )
+            val titleTextColor =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.TITLE_TEXT_COLOR)
+            val expandedBodyTextColor =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.EXPANDED_BODY_TEXT_COLOR)
+            val smallIcon = intentExtras.getString(PushTemplateConstants.IntentKeys.SMALL_ICON)
+            val smallIconColor =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.SMALL_ICON_COLOR)
+            val largeIcon = intentExtras.getString(PushTemplateConstants.IntentKeys.LARGE_ICON)
+            val customSound = intentExtras.getString(PushTemplateConstants.IntentKeys.CUSTOM_SOUND)
+            val actionButtonsString =
+                intentExtras.getString(PushTemplateConstants.IntentKeys.ACTION_BUTTONS_STRING)
+            val ticker = intentExtras.getString(PushTemplateConstants.IntentKeys.TICKER)
+            val tag = intentExtras.getString(PushTemplateConstants.IntentKeys.TAG) as String
+            val sticky = intentExtras.getBoolean(PushTemplateConstants.IntentKeys.STICKY)
+            val channelIdToUse = createChannelAndGetChannelID(
+                context, channelId, customSound, importance
+            )
```

Comment:
These values and the features are being used in basic, manual and filmstrip templates. Let discuss in our meeting tomorrow and make it such that this is reusable where possible.